src/FieldWarning/Assets/Model/Game/Loading.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Threading;
+using UnityEngine;
+
+public abstract class Loading
```

Comment:
I would prefer to use composition instead of inheritance here. That's to say I'd prefer to have this as a normal class and have TerrainMap hold an instance of it as a variable. The calls below would then look like

`_loader.AddWorker(LoadHeightMap, "Loading height map");`

which I think is a lot clearer. In my experience inheritance hierarchies are just very hard to understand and change in the future. But you have worked more with this code so I will defer to your opinion if you strongly disagree. 

It would also help a lot to have comments for the class and methods about what their role is.

Reply:
not a problem